pandas/io/parsers/c_parser_wrapper.py

```diff
@@ -34,7 +45,7 @@ class CParserWrapper(ParserBase):
 
     def __init__(
         self, src: FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str], **kwds
-    ):
+    ) -> None:
```

Comment:
`None` is optional for `__init__`.

Personally, I still like to add it. If mypy sees `__init__` without `None` as partially typed (I'm not sure whether that is the case), it would be good to keep `None`.

Reply:
I prefer adding it to

Reply:
we have some (now outdated) style guidelines for the typing in https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#style-guidelines.

The not including the return type of `__init__` was an unwritten style choice (IIRC preferred by @WillAyd)

If this is to be changed, we should be consistent and add it elsewhere (and update the style guidlines). I did have no strong preference originally but now that we have been not adding the redundant `None` for so long, I would now prefer the status quo.

Reply:
Ok, removed it